### PR TITLE
fix: remove unnecessary title attributes from datagrid

### LINKED ISSUE #5351: DataGrid : Option to disable/enable  datagrid tooltip
### Package

Carbon for IBM Products

### Description

By default the DataGrid adds  tooltip for the table header and table body.
In situations when the column content is not a plain text but a JSX element , the tooltip displays incorrect value 

We can either
- Disable the tooltip for the required column or Provide a custom string whenever a cell value is a JSX element /component

Screenshot 2024-05-28 at 8 14 34 PM

### Component(s) impacted

DataGrid 

### Browser

Chrome, Safari, Microsoft Edge

### @carbon/ibm-products (previously @carbon/ibm-cloud-cognitive) version

"@carbon/ibm-products": "2.40.0",

### Severity

Severity 2 = Aspects of design is broken, and impedes users in a significant way, but there is a way to complete their tasks. Affects major functionality, has a workaround.

### Product/offering

Infrastructure 

### CodeSandbox or Stackblitz example

https://codesandbox.io/p/sandbox/datagrid-clear-fliter-issue-forked-plpt64?file=%2Fsrc%2FExample.js%3A161%2C7

### Steps to reproduce the issue (if applicable)

_No response_

### Release date (if applicable)

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate issues

### PR DESCRIPTION
**Pull request by davidmenendez**
Closes #5351 

this PR removes a bunch of unnecessary `title` attributes that were strung throughout the `Datagrid` component that was causing accessibility issues. [current documentation](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/title#accessibility_concerns) tells us there's only a few valid instances in which to use `title` and none of the instances here meet the criteria.

> Outside of potential uses on input elements, there are notable instances where the title is the recommended attribute for accessibility user experiences.

### CODE REVIEW

File: packages/ibm-products/src/components/Datagrid/Datagrid/DatagridContent.tsx [hunk 1]
```diff
@@ -237,7 +236,6 @@ export const DatagridContent = ({
           [`${blockClass}__with-pagination`]: DatagridPagination,
           [`${blockClass}__dense-header`]: useDenseHeader,
         })}
-        title={gridTitle}
```
Comment (matthewgallo) [inline packages/ibm-products/src/components/Datagrid/Datagrid/DatagridContent.tsx:240]: This prop isn't a `title` attribute but an actual prop for the `TableContainer` from `@carbon/react` which displays an optional title for the table.
Screenshot 2024-06-24 at 3 46 00 PM

Comment (davidmenendez) [reply packages/ibm-products/src/components/Datagrid/Datagrid/DatagridContent.tsx:240]: oooh i see. ok. my mistake. will revert those changes.

**Comment by davidmenendez** [comment]
@matthewgallo thanks for catching that title issue. i went ahead and reverted that commit and only removed instances related to `title`.

i also added an additional small fix that adds `role: undefined` to cells and rows to alleviate violations that were showing up in the accessibility checker while testing

Screenshot 2024-06-24 at 3 26 42 PM

took us down from 209 to 59 violations. progress 😅 

Screenshot 2024-06-24 at 3 25 14 PM
Screenshot 2024-06-24 at 3 25 19 PM

**Comment by davidmenendez** [comment]
the last commit i added removed some additional problematic roles. this gets our violation count down to 32 💪

**Comment by davidmenendez** [comment]
actually went ahead and addressed #4032 in the last commit because the fix was basically the same

**Review by matthewgallo — APPROVED** [review]

**Pull request merged**